src/core/renderers/canvas/utils/CanvasMaskManager.js

```diff
@@ -153,12 +208,63 @@ export default class CanvasMaskManager
     }
 
     /**
+     * Makes the black on the giben context transparent. This is necessary to make this canvas mask act as the
+     * webgl version with black masks.
+     * @param {Context} context - The context in which we want to convert black pixels into transparent.
+     */
+    _makeBlackTransparent(context)
+    {
+        const imgd = context.getImageData(0, 0, context.canvas.width, context.canvas.height);
+        const pix = imgd.data;
+
+        for (let i = 0, n = pix.length; i < n; i += 4)
+        {
+            if (pix[i] === 0 && pix[i + 1] === 0 && pix[i + 2] === 0)
+            {
+                pix[i + 3] = 0;
+            }
+        }
+
+        context.putImageData(imgd, 0, 0);
+    }
+
+    /**
      * Restores the current drawing context to the state it was before the mask was applied.
      *
      * @param {PIXI.CanvasRenderer} renderer - The renderer context to use.
      */
     popMask(renderer)
     {
+        this._freeCanvasOnNextFrame = this._freeCanvasOnNextFrame || [];
+        this._freeCanvasOnNextFrame.forEach((canvas) => this._pool.freeElement(canvas));
```

Comment:
Need to set `this._freeCanvasOnNextFrame.length = 0`, or you are going to free extra canvases more than likely.